### PR TITLE
Clarify that `ENetConnection`'s `create_host` and `service` must be called on client and server

### PR DESCRIPTION
**Pull request by marcospb19**
Context: despite 3 years working coding servers and clients, it took me around **10 hours** to connect two `ENet` peers, and I read docs + ENet website + searched in issues and forums (oof!).

So, in this PR I try to clarify some of things I wish were there.

(In a separate PR I plan on adding an example.)

### CODE REVIEW

File: modules/enet/doc_classes/ENetConnection.xml
Comment (marcospb19) [inline modules/enet/doc_classes/ENetConnection.xml:65]: to be honest, this one wasn't a pain point, but I thought it could be useful information.

File: modules/enet/doc_classes/ENetConnection.xml
Comment (marcospb19) [inline modules/enet/doc_classes/ENetConnection.xml:78]: slight reword, also not a pain point, I can revert if u want :)

Comment (Mickeon) [reply modules/enet/doc_classes/ENetConnection.xml:78]: [We tend to avoid "but" when possible](https://docs.godotengine.org/en/stable/contributing/documentation/docs_writing_guidelines.html#ban-these-8-words).

Comment (marcospb19) [reply modules/enet/doc_classes/ENetConnection.xml:78]: I gave it a second and third read _but_ couldn't find any problematic nuance on this specific one :monocle_face:.

`but` is used there to contrast `create_host` with `create_host_bound`, I'm open to any suggestions tho :) .

Comment (Mickeon) [reply modules/enet/doc_classes/ENetConnection.xml:78]: I am personally not a big fan of "see another method first to know what this method does". Ideally, method descriptions should be self-explanatory. In every other PR of mine I replaced these kinds of descriptions with proper explanations.

In this case I think it would be fine to repeat the description of `create_host` and then state the difference between the two.

Comment (marcospb19) [reply modules/enet/doc_classes/ENetConnection.xml]: thanks for explaining :) I pushed a commit (yall squash, right?) that goes through all parameters in order instead of pointing to another method

File: modules/enet/doc_classes/ENetConnection.xml
Comment (Mickeon) [inline modules/enet/doc_classes/ENetConnection.xml:65]:
```suggestion
				This method binds a randomly numbered UDP port on the host machine at the [i]unspecified[/i] address. Use [method create_host_bound] to specify the address and port.
```

File: modules/enet/doc_classes/ENetConnection.xml
Comment (Mickeon) [inline modules/enet/doc_classes/ENetConnection.xml:149]:
```suggestion
				[b]Note:[/b] This method must be called on both ends involved in the event (sending and receiving hosts).
```

**Review by Mickeon — APPROVED** [review]
Docs are pretty dang good now, assuming they are accurate.

Also remember to squash your commits, see [the Pull Request Workflow](https://docs.godotengine.org/en/stable/contributing/workflow/pr_workflow.html#the-interactive-rebase).

**Comment by ze2j** [comment]
I may be wrong as I am not an english native speaker, but I think _stablish_ should be changed to _establish_ in the notes.

File: modules/enet/doc_classes/ENetConnection.xml
Comment (Mickeon) [inline modules/enet/doc_classes/ENetConnection.xml:66]:
```suggestion
				[b]Note:[/b] It is necessary to create a host in both client and server in order to establish a connection.
```

File: modules/enet/doc_classes/ENetConnection.xml
Comment (Mickeon) [inline modules/enet/doc_classes/ENetConnection.xml:79]:
```suggestion
				[b]Note:[/b] It is necessary to create a host in both client and server in order to establish a connection.
```

**Review by Mickeon — COMMENTED** [review]
@ze2j is right. Just putting these as additional suggestions so that they can be picked more easily.

**Comment by marcospb19** [comment]
@Mickeon 
> Also remember to squash your commits, see [the Pull Request Workflow](https://docs.godotengine.org/en/stable/contributing/workflow/pr_workflow.html#the-interactive-rebase).

Sure I can do it, but GitHub already enables maintainers to squash and merge, and in the process, even rewrite the commit messages if they want to.

I see some back and forth in this repo like "change commit message" or "rebase every commit", while I see the reasons behind it, is there any reason why maintainers don't prefer to click on "Squash and Merge" button? Wouldn't it save time for both the maintainers and contributors?

I also ask this because I believe that fixup commits can be useful for reviewers!

---

Anyway! I won't block the PR on this question :) , applying changes and squashing, I appreciate the review.

**Comment by marcospb19** [comment]
Another small argument: Git beginners can have a hard time with interactive rebasing, IMHO this adds an unnecessary barrier to contributing, considering there is a button that solves it (it adds two more clicks, max).

**Comment by marcospb19** [comment]
Done, applied all changes, ty @ze2j too!!!

**Comment by AThousandShips** [comment]
> GitHub already enables maintainers to squash and merge

It does, but sadly it doesn't allow squashing *and* creating a dedicated merge commit, which we require for any PR to make the git history cleaner and more manageable, it also helps reverting changes, and helps keeping the history clean in the rare cases where we *don't* enforce single commits, having merge commits also helps with cherry picking I believe

And rebasing and squashing is a core part of the workflow in any repo and any project and is a good and important skill to know, so at least I feel that it's less of a barrier and more a part of the general process of learning how to work with the tools and workflows necessary for contributing, just as much as learning how to make commits themselves or create branches! And maintainers are ready to help if new contributors are having a hard time and ask for help to figure it out, along with the detailed documentation

File: modules/enet/doc_classes/ENetConnection.xml
Comment (mhilbrunner) [inline modules/enet/doc_classes/ENetConnection.xml:65]:
```suggestion
				This method binds a random available dynamic UDP port on the host machine at the [i]unspecified[/i] address. Use [method create_host_bound] to specify the address and port.
```
Not sure if this distinction matters, but it is guaruanteed to be a currently unused port in the dynamic/private/ephemeral port range (49152 to 65535).

**Review by mhilbrunner — APPROVED** [review]
Thanks for adding these clarifications, looks good to me!

File: modules/enet/doc_classes/ENetConnection.xml
Comment (marcospb19) [reply modules/enet/doc_classes/ENetConnection.xml:65]: Woow nice catch! I'll apply it.

**Comment by marcospb19** [comment]
@mhilbrunner

> > GitHub already enables maintainers to squash and merge
> 
> It does (though it's not enabled on this repo), but sadly it doesn't allow squashing _and_ creating a dedicated merge commit, which we require for any PR to make the git history cleaner and more manageable, it also helps reverting changes, and helps keeping the history clean in the rare cases where we _don't_ enforce single commits, having merge commits also helps with cherry picking I believe
> 
> And rebasing and squashing is a core part of the workflow in any repo and any project and is a good and important skill to know, so at least I feel that it's less of a barrier and more a part of the general process of learning how to work with the tools and workflows necessary for contributing, just as much as learning how to make commits themselves or create branches! And maintainers are ready to help if new contributors are having a hard time and ask for help to figure it out, along with the detailed documentation

Oh, I didn't know merge commits helped with cherry-picking and reverting changes! That's great to know.

Thanks for clarifying :) 

---

CI [failed](https://github.com/godotengine/godot/actions/runs/9884087929/job/27299918100?pr=93987) after I applied [this suggestion](https://github.com/godotengine/godot/pull/93987#discussion_r1670838030), it seems unrelated tho.

**Comment by akien-mga** [comment]
Thanks!

**Comment by Jimmio92** [comment]
You don't call create_host on clients -- that's what you call to start serving a session. If the client is the server, they can't be a client to someone else, so maybe this isn't worded right?

You do need to create an ENetConnection on all sides, client and server, but you don't call `create_host` on clients, you call `create_client`.

If for some reason you needed to create a host on clients to get connections to work, that's a bug imo.

**Comment by Faless** [comment]
> You do need to create an ENetConnection on all sides, client and server, but you don't call `create_host` on clients, you call `create_client`.

@Jimmio92 You are confusing [`ENetConnection`](https://docs.godotengine.org/en/stable/classes/class_enetconnection.html#class-enetconnection) with [`ENetMultiplayerPeer`](https://docs.godotengine.org/en/stable/classes/class_enetmultiplayerpeer.html).
The first one, does indeed need either `create_host` or `create_host_bound`, and doesn't have a `create_client` method since it's almost a one-to-one mapping to official ENet API.

**Comment by Jimmio92** [comment]
Ahh, I now see I absolutely did confuse the two. Sorry about that!

**Pull request merged**
